palantir-java-format/src/main/java/com/palantir/javaformat/java/java14/Java14InputAstVisitor.java

```diff
@@ -226,12 +231,30 @@ public Void visitCase(CaseTree node, Void unused) {
                 token("-");
                 token(">");
                 builder.space();
-                scan(node.getBody(), null);
+                if (node.getBody().getKind() == BLOCK) {
+                    visitBlock(
```

Comment:
For collapsing empty case blocks: `case ENUM -> {}`

Reply:
Mind adding this as a source comment? :-)

Reply:
Added the comment 👍